kernel/smp.c

```diff
@@ -17,23 +17,26 @@ static atomic_t start_flag;
 unsigned int z_smp_global_lock(void)
 {
 	unsigned int key = arch_irq_lock();
+	struct k_thread *const current = _current;
```

Comment:
This code is the legacy irq_lock() emulation that is only there for compatibility.  Not worth performance work IMHO.